src/app/ui/dynamics_popup.cpp

```diff
@@ -293,9 +295,9 @@ int DynamicsPopup::ditheringIndex() const {
   return 0;
 }
 
-void DynamicsPopup::loadDynamicPref(ToolPreferences* toolPref) {
-  if (toolPref) {
-    auto& dynaPref = toolPref->dynamics;
+void DynamicsPopup::loadDynamicsPref(Preferences* pref) {
+  if (pref) {
+    auto& dynaPref = pref->tool(App::instance()->activeTool()).dynamics;
```

Comment:
No need to pass the singleton Preferences instance as argument:
```suggestion
void DynamicsPopup::loadDynamicsPref()
{
  auto& dynaPref = Preferences::instance().tool(App::instance()->activeTool()).dynamics;
```